package_nso_to_oc/xe/xe_routing_policy.py

```diff
@@ -0,0 +1,521 @@
+        all_processed = True
+
+        for seq_index, seq in enumerate(prefix.get("seq", [])):
+            if "deny" in seq:
+                all_processed = False
+                routing_policy_notes.append(
+f"""
+Prefix Name: {prefix.get("name")}
+Sequence Number: {seq.get("no")}
+This sequence contains a deny operation, which is not supported in OpenConfig. Translation to OC was skipped.
+""")
+                continue
+            if not "permit" in seq:
+                continue
+
+            masklength = "exact" if not "le" in seq["permit"] else f'{seq["permit"]["ge"]}..{seq["permit"]["le"]}'
+            new_prefix = {
+                "openconfig-routing-policy:ip-prefix": seq["permit"].get("ip"),
+                "openconfig-routing-policy:masklength-range": masklength,
+                "openconfig-routing-policy:config": {
+                    "openconfig-routing-policy:ip-prefix": seq["permit"].get("ip"),
+                    "openconfig-routing-policy:masklength-range": masklength,
+                    "openconfig-routing-policy-ext:seq": seq["no"]
+                }
+            }
+            prefixes.append(new_prefix)
+            
+            # Ensure the value we're nullifying does exist
+            if common.get_index_or_default(seq_list_after, seq_index, None):
+                seq_list_after[seq_index] = None
+        
+        prefix_sets["openconfig-routing-policy:prefix-sets"]["openconfig-routing-policy:prefix-set"].append(new_prefix_set)
+
+        if all_processed:
+            common.get_index_or_default(xe_prefixes_after, prefix_index, {})["name"] = None
+    
+    openconfig_routing_policies["openconfig-routing-policy:routing-policy"]["openconfig-routing-policy:defined-sets"].update(prefix_sets)
+
+def process_as_path_sets(config_before, config_after):
+    as_path_sets = {"openconfig-bgp-policy:as-path-sets": {"openconfig-bgp-policy:as-path-set": []}}
+    access_list = config_before.get("tailf-ned-cisco-ios:ip", {}).get("as-path", {}).get("access-list", [])
+    access_list_after = config_after.get("tailf-ned-cisco-ios:ip", {}).get("as-path", {}).get("access-list", [])
+    all_processed = True
+
+    for access_index, access in enumerate(access_list):
+        new_path_set = {
+            "openconfig-bgp-policy:as-path-set-name": access.get("name"),
+            "openconfig-bgp-policy:config": {
+                "openconfig-bgp-policy:as-path-set-name": access.get("name"),
+                "openconfig-bgp-policy:as-path-set-member": []
+            }
+        }
+        members = new_path_set["openconfig-bgp-policy:config"]["openconfig-bgp-policy:as-path-set-member"]
+        rule_list_after = common.get_index_or_default(access_list_after, access_index, {}).get("as-path-rule", [])
+
+        for rule_index, rule in enumerate(access.get("as-path-rule", [])):
+            if not "operation" in rule:
+                continue
+            if rule["operation"] == "deny":
```

Comment:
If deny in as-path-set, we must note, keep processing for other notes, but as-path-sets missing statements can't be appended to OC prefix lists.